_rules/SC1-1-1-accessible-name-is-image-filename.md

```diff
@@ -0,0 +1,88 @@
+---
+name: Image accessible name is filename
+description: |
+  This rule checks that image elements that use their source filename as their accessible name do so without loss of infomation to the user.
+
+success_criterion:
+- 1.1.1 # Non-Text Content
+
+test_aspects:
+- DOM Tree
+- CSS Styling
+
+authors:
+- Bryn Anderson
+---
+
+## Test procedure
+
+### Applicability
+
+The rule applies to HTML `input` element with a [`type`](https://www.w3.org/TR/html/sec-forms.html#dom-htmlinputelement-type) of `image`, or `area` elements, or any [non-decorative](#decorative) HTML element with the semantic role of `img`, where the element is [included in the accessibility tree](#included-in-the-accessibility-tree) and has an [accessible name](#accessible-name) that is equivalent to the [filename](#filename) specified in the `src` attribute. Difference in letter casing, and forward and trailing whitespace should be ignored. 
+
+### Expectation
+
+Each test target has an accessible name that serves an equivalent purpose can of the [non-text content](https://www.w3.org/TR/WCAG21/#dfn-non-text-content).
+
+## Assumptions
+
+*There are currently no assumptions*
+
+## Accessibility support
+
+ *There are no major accessibility support issues known for this rule.*
+
+## Background
+
+- [Understanding Success Criterion 1.1.1: Non-text Content](https://www.w3.org/WAI/WCAG21/Understanding/non-text-content.html)
+- [F30: Failure of Success Criterion 1.1.1 and 1.2.1 due to using text alternatives that are not alternatives (e.g., filenames or placeholder text)](https://www.w3.org/WAI/WCAG21/Techniques/failures/F30)
+
+## Test Cases
+
+### Pass example 1
+
+The `img` element's accessible name uses the filename and accurately describes the image.
+
+```html
+<img src="https://www.w3.org/WAI/demos/bad/img/w3c" alt="w3c">
+```
+
+### Failure example 1
+
+
+The `img` element's accessible name matches the image filename which does not acurately describe the image.
+
+```html
+<img src="https://www.w3.org/WAI/demos/bad/after/img/teaser_right2.jpg" alt="teaser_right2.jpg">
+```
+
```

Comment:
I think these examples would benefit from a failure example that explicitly makes it clear that extensions should never be part of a text alternative because extensions do not serves the same purpose as the image (if my reading of the expectation is correct).

Suggestion:

### Failure example 2
The `img` element's accessible name matches the image filename which has a file extension which doesn't serves the same purpose as the image.

```html
<img src="https://www.w3.org/WAI/demos/bad/img/w3c.png" alt="w3c.png">
```

Reply:
I do not agree ... Relevance of a text alternative is so unpredictable. 

You don't know if "W3C.png" would never be a conforming text alternative in a particular case, nor if a perfect matching between a src and a alt (including an extension) can't never be a passed test. 

> because extensions do not serves the same purpose as the image

I don't think it can be so systematic. 

Imagine a photo gallery application (like a media library in a content management system), which the only contentS are photos. In that case, for a blind user that just uploaded a photo in the library, the important information is the the name of the filename (to reach the one he has just uploaded). 
The alt attribut could be use to bring that information (and as an expert in that particular context of use, I won't considered that as a failure). 
In that case, the alt is not use to describe the image, but to inform about the name of the file. 

So i think "Each test target has an accessible name that serves an equivalent purpose " is a wide enough to cover each case (describe what is on the image, describe the function of the image or other use case we don't know yet). 

Sorry for having beeing so long, I'm done now :-)

Reply:
That is a good point. I guess there are exceptions. I think an example where that is not the case (like my example) is still a good addition but I now agree that my generalizing statement about file extensions never serving the same purpose as the image is incorrect.